### PR TITLE
add origin application version to header

### PR DESCRIPTION
**Pull request by ohadbitt**
add origin application version to connectionStringBuilder to be added to the header (if not given will use the old way to retrieve java client version)

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 1]
```diff
@@ -19,10 +19,21 @@
 
     private AadAuthenticationHelper aadAuthenticationHelper;
     private String clusterUrl;
+    private String originApplicationVersion;
```
Comment (toshetah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: ClientVersionForTracing

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java [hunk 2]
```diff
@@ -15,13 +15,15 @@
     private X509Certificate x509Certificate;
     private PrivateKey privateKey;
     private String aadAuthorityId; // AAD tenant Id (GUID)
+    private String originApplicationVersion;
```
Comment (toshetah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java]: ClientVersionForTracing, and it should be all along.

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java [hunk 3]
```diff
@@ -94,6 +96,17 @@ public static ConnectionStringBuilder createWithAadApplicationCredentials(String
         return createWithAadApplicationCredentials(resourceUri, applicationClientId, applicationKey, null);
     }
 
+    public static ConnectionStringBuilder createWithAadApplicationCredentials(String resourceUri,
+                                                                              String applicationClientId,
+                                                                              String applicationKey,
+                                                                              String authorityId,
+                                                                              String originApplicationVersion)
+    {
```
Comment (toshetah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java]: originApplicationVersion is not related to authentication.
Please add a setter and use it in order to change it.

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java (hunk 1 above)
Comment (toshetah) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: It is not resolved.

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 4]
```diff
@@ -19,10 +19,21 @@
...
 
     public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
         clusterUrl = csb.getClusterUrl();
         aadAuthenticationHelper = new AadAuthenticationHelper(csb);
+        if(csb.originApplicationVersion() == null){
+            String version = Utils.class.getPackage().getImplementationVersion();
```
Comment (toshetah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: What is that? Utils.class.getPackage().getImplementationVersion();

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 5]
```diff
@@ -19,10 +19,21 @@
...
+            originApplicationVersion = "Kusto.Java.Client";
+            if (StringUtils.isNotBlank(version)) {
+                originApplicationVersion += ":" + version;
+            }
+        }
+        else{
+            originApplicationVersion = csb.originApplicationVersion();
```
Comment (toshetah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: ClientVersionForTracing

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 6]
```diff
@@ -74,6 +85,6 @@ public Results execute(String database, String command, ClientRequestProperties
             throw new DataClientException(clusterEndpoint, String.format(clusterEndpoint, "Error in executing command: %s, in database: %s", command, database), e);
         }
 
-        return Utils.post(clusterEndpoint, aadAccessToken, jsonString, timeoutMs.intValue());
+        return Utils.post(clusterEndpoint, aadAccessToken, jsonString, timeoutMs.intValue(), originApplicationVersion);
```
Comment (toshetah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: ClientVersionForTracing

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java [hunk 7]
```diff
@@ -15,13 +15,15 @@
...
     private String aadAuthorityId; // AAD tenant Id (GUID)
+    private String originApplicationVersion;
 
     String getClusterUrl() { return clusterUri; }
     String getUserUsername() { return username; }
     String getUserPassword() { return password; }
     String getApplicationClientId() { return applicationClientId; }
     String getApplicationKey() { return applicationKey; }
     String getAuthorityId() { return aadAuthorityId; }
+    String originApplicationVersion() { return originApplicationVersion; }
```
Comment (toshetah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java]: ClientVersionForTracing

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java (hunk 3 above)
Comment (toshetah) [reply data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java]: Please remove originApplicationVersion from the constructor.

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java (hunk 1 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: yea idk why it didnt go in the same commit

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java (hunk 4 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: its exactly what was used before in Utils.java 
its looking at the version in the module manifest

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 8]
```diff
@@ -19,10 +19,21 @@
...
         aadAuthenticationHelper = new AadAuthenticationHelper(csb);
+        if(StringUtils.isNotBlank(csb.getClientVersionForTracing()) ){
+            clientVersionForTracing = "Kusto.Java.Client";
+            String version = Utils.GetPackageVersion();
+            if (StringUtils.isNotBlank(version)) {
+                clientVersionForTracing += ":" + version;
+            }
+        }
+        else{
+            clientVersionForTracing = csb.getClientVersionForTracing();
```
Comment (toshetah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: if getClientVersionForTracing is blank, don't you want to assign something else?

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java [hunk 9]
```diff
@@ -126,4 +155,11 @@ public static ConnectionStringBuilder createWithAadApplicationCertificate(String
         return csb;
     }
 
+    public void setClientVersionForTracing(String clientVersionForTracing) {
```
Comment (tamirkamara) [inline data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java]: Have you decided to keep this?
Please move this up, next to the right getter

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java [hunk 10]
```diff
@@ -126,4 +155,11 @@ public static ConnectionStringBuilder createWithAadApplicationCertificate(String
...
+        this.clientVersionForTracing = clientVersionForTracing;
+    }
+
+    public void setApplicationNameForTracing(String applicationNameForTracing) {
```
Comment (tamirkamara) [inline data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java]: Please move this up, next to the right getter

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 11]
```diff
@@ -19,10 +19,22 @@
...
     public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
         clusterUrl = csb.getClusterUrl();
         aadAuthenticationHelper = new AadAuthenticationHelper(csb);
+        clientVersionForTracing = "Kusto.Java.Client";
+        String version = Utils.GetPackageVersion();
+        if (StringUtils.isNotBlank(version)) {
+            clientVersionForTracing += ":" + version;
+        }
+
+        if (StringUtils.isNotBlank(csb.getClientVersionForTracing())) {
```
Comment (tamirkamara) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: Do we still need that if the user is going to use the applicationNameForTracing ?

Comment (toshetah) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: Yes. We need to modify it from the ingest client.

**Pull request merged**